packages/wrangler/src/config/index.ts

```diff
@@ -324,7 +324,7 @@ export function printBindings(bindings: CfWorkerInit["bindings"]) {
 			type: "Unsafe Metadata",
 			entries: Object.entries(unsafe.metadata).map(([key, value]) => ({
 				key,
-				value: `${value}`,
+				value: `${typeof value === "object" ? JSON.stringify(value) : value}`,
```

Comment:
Perhaps just do `JSON.stringify()` for all values?

Reply:
I'll add some tests for nested objects (just spotted the existing ones only test strings)

Not sure if the indentation will work that nicely with `null, 2` – because of the hyphen dot points. Will check.

Had a look at unsafe bindings – they use the name of the param, eg:
```
Your worker has access to the following bindings:
- Unsafe:
  - origin: MY_ORIGIN_NAME
  - origin: MY_OTHER_ORIGIN_NAME
```

Given the shape of unsafe metadata is arbitrary, there's not much we can copy there. So JSON might be the best bet.

Reply:
> I'll add some tests for nested objects (just spotted the existing ones only test strings)
Thanks for spotting that and adding tests for this 😃